plugins/module_utils/network/sonic/config/dhcp_relay/dhcp_relay.py

```diff
@@ -0,0 +1,490 @@
+
+        return requests
+
+    def get_delete_specific_dhcp_relay_param_requests(self, command, config):
+        """Get requests to delete specific DHCP relay configurations
+        based on the command specified for the interface
+        """
+        requests = []
+
+        name = command['name']
+        ipv4 = command.get('ipv4')
+        have_ipv4 = config.get('ipv4')
+        if not ipv4 or not have_ipv4:
+            return requests
+
+        server_addresses = self.get_server_addresses(ipv4.get('server_addresses'))
+        have_server_addresses = self.get_server_addresses(have_ipv4.get('server_addresses'))
+
+        # Delete all DHCP relay config for an interface, if only
+        # an address with no value is specified.
+        if (ipv4.get('server_addresses') and len(ipv4.get('server_addresses'))
+                and not server_addresses):
+            requests.append({'path': self.dhcp_relay_intf_config_path['server_addresses_all'].format(intf_name=name), 'method': DELETE})
+            return requests
+
+        del_server_addresses = have_server_addresses.intersection(server_addresses)
+        if del_server_addresses:
+            # Delete all DHCP relay config for an interface if all
+            # server addresses are to be deleted
+            if len(del_server_addresses) == len(have_server_addresses):
+                requests.append({'path': self.dhcp_relay_intf_config_path['server_addresses_all'].format(intf_name=name), 'method': DELETE})
+                return requests
+
+            for addr in del_server_addresses:
+                url = self.dhcp_relay_intf_config_path['server_address'].format(intf_name=name, server_address=addr)
+                requests.append({'path': url, 'method': DELETE})
+
+        # Specifying appropriate order for deletion to succeed
+        if ipv4.get('link_select') is not None and have_ipv4.get('link_select'):
+            url = self.dhcp_relay_intf_config_path['link_select'].format(intf_name=name)
+            requests.append({'path': url, 'method': DELETE})
+
+        if ipv4.get('source_interface') and have_ipv4.get('source_interface'):
+            url = self.dhcp_relay_intf_config_path['source_interface'].format(intf_name=name)
+            requests.append({'path': url, 'method': DELETE})
+
+        if (ipv4.get('max_hop_count') and have_ipv4.get('max_hop_count')
+                and have_ipv4.get('max_hop_count') != DEFAULT_MAX_HOP_COUNT):
+            url = self.dhcp_relay_intf_config_path['max_hop_count'].format(intf_name=name)
+            requests.append({'path': url, 'method': DELETE})
+
+        if ipv4.get('vrf_select') is not None and have_ipv4.get('vrf_select'):
+            url = self.dhcp_relay_intf_config_path['vrf_select'].format(intf_name=name)
+            requests.append({'path': url, 'method': DELETE})
+
+        if (ipv4.get('policy_action') and have_ipv4.get('policy_action')
+                and have_ipv4.get('policy_action') != DEFAULT_POLICY_ACTION):
```

Comment:
Please verify matching "have" and "want" configuration before enqueuing the request.

Reply:
Done